### ISSUE TITLE
Trilinos: Update PR Template to Reflect SQA Needs

### ISSUE THREAD

**Issue by csiefer2** [trilinos/Trilinos#5979]
The recent SQA exercise seems to exposed some questions we're not asking that we probably should (and questions we are asking that we don't need to).

@trilinos/tpetra @trilinos/muelu @trilinos/framework

**Comment by csiefer2** [comment]
Current PR template:


@trilinos/<teamName>

## Description


## Motivation and Context




## How Has This Been Tested?






## Checklist

- [ ] My commit messages mention the appropriate GitHub issue numbers.
- [ ] My code follows the code style of the affected package(s).
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the [code contribution guidelines](../blob/master/CONTRIBUTING.md) for this project.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [ ] No new compiler warnings were introduced.
- [ ] These changes break backwards compatibility.

**Comment by csiefer2** [comment]
As a first cut, I propose:
```
<!---
Be sure to select `develop` as the `base` branch against which to create this
pull request.  Only pull requests against `develop` will undergo Trilinos'
automated testing.  Pull requests against `master` will be ignored.

Provide a general summary of your changes in the Title above.  If this pull
request pertains to a particular package in Trilinos, it's worthwhile to start
the title with "PackageName:  ".

Note that anything between these delimiters is a comment that will not appear
in the pull request description once created. Most areas in this message are
commented out and can be easily added by removing the comment delimiters.

Please make sure to mark:
* Reviewers
* Assignees
* Labels

Replace <teamName> below with the appropriate Trilinos package/team name.
-->
@trilinos/<teamName>

## Motivation
<!--- Why is this change required?  What problem does it solve? Feel free to link to a github issue that describes the problem, if one exists.-->

## Stakeholder Feedback
<!--- If a github issue includes feedback from the relevant stakeholder(s), please link it.  If the stakeholder(s) communicated that feedback through a different medium, please note that you did so. -->


## Testing
<!---
Please confirm that any classes or functions in the Trilinos library that this PR touches are exercised  by at least one test in Trilinos.  For untestable changes (e.g. changes to the nightly testing system) or changes to Trilinos tests, please say "N/A".-->

<!--- 
## Additional Information
Anything else we need to know in evaluating this merge request?
 -->

**Comment by csiefer2** [comment]
The big changes in the proposed template are:
a) Adding a section for stakeholder feedback
b) Being explicit in the instructions as to what we want to see in the 'testing' section
c) Getting rid of the checkboxes that everyone ignores anyway

**Comment by jhux2** [comment]
@csiefer2 Thank you for opening this issue.  We need to have this discussion.

> The big changes in the proposed template are:
> a) Adding a section for stakeholder feedback
> b) Being explicit in the instructions as to what we want to see in the 'testing' section
> c) Getting rid of the checkboxes that everyone ignores anyway

I agree.  Additional comments:
   - I dislike having instruction in comments because they aren't visible w/o editing. Imo, they should be moved into text blocks, maybe italicized or something.
   - Be more assertive in requiring a supporting issue:
`Provide a link to the github issue that describes the problem.  It should explain why this change is required and what problem/issue/bug the change solves. `
   - I think the "related issues" section has value.

**Comment by csiefer2** [comment]
>     * Be more assertive in requiring a supporting issue:

Why?  PRs *are* issues.

**Comment by tjfulle** [comment]
I agree with @jhux2 RE supporting issues.  I don’t view PRs as issues but, rather, resolutions to open issues.  A successful PR should close 1 or more issues.

**Comment by csiefer2** [comment]
> Technically speaking, yes. But they're not always treated that way -- I've seen plenty without comments (my own included).

Sure.  So, if you've got an issue link it.  If the PR is the issue, then you don't.

**Comment by csiefer2** [comment]
@kddevin Thoughts?

**Comment by jhux2** [comment]
> Why? PRs _are_ issues.

After more thought, I disagree.  Search for issues requires "is:issue", versus "is:pr" for PRs.  PRs are about merging code.  Sure, one can discuss whatever in the PR, but one would not report a bug/problem, request an enhancement, or propose a redesign in a PR.

**Comment by kddevin** [comment]
I agree that a separate issue is helpful in almost all cases, for exactly the reason stated by @jhux2 above.  When I was searching for examples for SQA, it was frustrating to search issues for something I *knew* existed, only to realize after not finding it that it was a PR.  So in almost all cases, having both would be helpful.  That said, I will have to change my normal mode of operation, as I was guilty of creating only PRs.

Having the Related Issues section is really important.  I use the related issues links all the time, not only at SQA time.

The checklists are almost always ignored, and I never cited them in the SQA.  I am OK with removing them.

For testing, I'd ask for more than confirmation.  I'd ask for the names of the tests that exercise the new capability and the platforms on which the code was built and the tests were run.

**Comment by csiefer2** [comment]
Proposed update:
```
<!---
Be sure to select `develop` as the `base` branch against which to create this
pull request.  Only pull requests against `develop` will undergo Trilinos'
automated testing.  Pull requests against `master` will be ignored.

Provide a general summary of your changes in the Title above.  If this pull
request pertains to a particular package in Trilinos, it's worthwhile to start
the title with "PackageName:  ".

Note that anything between these delimiters is a comment that will not appear
in the pull request description once created. Most areas in this message are
commented out and can be easily added by removing the comment delimiters.

Please make sure to mark:
* Reviewers
* Assignees
* Labels

Replace <teamName> below with the appropriate Trilinos package/team name.
-->
@trilinos/<teamName>

## Motivation
<!--- Why is this change required?  What problem does it solve? Please link to a github issue that describes the problem/issue/bug this PR solves.
-->

<!---
If applicable, let us know how this merge request is related to any other open
issues or pull requests:

## Related Issues

* Closes 
* Blocks 
* Is blocked by 
* Follows 
* Precedes 
* Related to 
* Part of 
* Composed of 
-->


## Stakeholder Feedback
<!--- If a github issue includes feedback from the relevant stakeholder(s), please link it.  If the stakeholder(s) communicated that feedback through a different medium, please note that you did so. -->

## Testing
<!---
Please confirm that any classes or functions in the Trilinos library that this PR touches are exercised  by at least one test in Trilinos.  Please specify which test that is.  For untestable changes (e.g. changes to the nightly testing system) or changes to Trilinos tests, please say "N/A".

-->

<!--- 
## Additional Information
Anything else we need to know in evaluating this merge request?
 -->

**Comment by csiefer2** [comment]
@rppawlo You also did the most recent SQA.  Any suggestions?
